### PR TITLE
[Localization] New Pipeline to replace github action - not from fork

### PR DESCRIPTION
**Pull request by tj-devel709**
* Creating this PR from this branch since forked repos cannot run the governance checks - [old PR](https://github.com/xamarin/xamarin-macios/pull/17848)

I created this Pipeline - [xamarin.macios-legoChanges](https://devdiv.visualstudio.com/DevDiv/_build?definitionId=18272&_a=summary) to replace the Github action that we cannot use anymore to carry over the changes from the Loc team. This pipeline only listens to the Localization branch, but we need it in main since the main branch overwrites the Localization branch regularly!

### CODE REVIEW

File: tools/devops/automation/build-lego.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        vmImage: windows-latest
+      steps:
+      - template: templates/governance-checks.yml
+        parameters:
+          isPR: false
+          repositoryAlias: self
+          commit: HEAD
+
+    - job: BringLocChanges
+      displayName: 'Copy over the lcl file changes from Localization branch'
```
Comment (rolfbjarne) [inline tools/devops/automation/build-lego.yml:72]: Is this supposed to be inside the `Governance Checks` stage? Or am I reading the yaml wrong?

File: tools/devops/automation/build-lego.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      pool:
+        vmImage: windows-latest
+      steps:
+
+      - pwsh: |
+          git remote remove origin
+          git remote add origin https://$(GitHub.Token)@github.com/xamarin/xamarin-macios.git
+          git remote # don't add -v else we see the pat
+
+          git config user.email "valco@microsoft.com"
```
Comment (rolfbjarne) [inline tools/devops/automation/build-lego.yml:82]: Should we use our team email instead?

File: tools/devops/automation/build-lego.yml (hunk 1 above)
Comment (tj-devel709) [reply tools/devops/automation/build-lego.yml:72]: This is how we had the loc stuff before - inside the Governance Stage but I'm not sure if there is a reason we need to have it inside. @mandel-macaque?

File: tools/devops/automation/build-lego.yml (hunk 2 above)
Comment (tj-devel709) [reply tools/devops/automation/build-lego.yml:82]: @mandel-macaque This is what we have elsewhere, thoughts?

**Review by rolfbjarne — APPROVED** [review]

**Review by haritha-mohan — APPROVED** [review]

**Pull request merged**